### ISSUE TITLE
[1.9] Server crash on Dark Tank

### ISSUE THREAD

**Issue by kwpugh** [rubensworks/EvilCraft#368]
Connecting a Dark Tank to a Purifier or trying to put fluids into it via bucket crashes the server.

Forge: 1887
Mod: EvilCraft-1.9-0.9.9.jar and CyclopsCore-1.9-0.5.8.jar

Crash Report:  http://pastebin.com/raw/6LVHJJG4

**Comment by MrThomson** [comment]
also sometime cause client crash and 

Server Crash Report 

---- Minecraft Crash Report ----

WARNING: coremods are present:
  FldLoadingPlugin (FastLeafDecay-1.9-1.6.jar)
  LoadingPlugin (RandomThings-MC1.9-3.7.jar)
  Reborn Core ASM (RebornCore-1.9-2.1.0.53-universal.jar)
  MalisisCorePlugin (malisiscore-1.9-3.0.6.jar)
  MalisisSwitchesPlugin (malisisswitches-1.9-2.0.0.jar)
Contact their authors BEFORE contacting forge

// There are four lights!

Time: 5/9/16 4:07 PM
Description: Ticking block entity

java.lang.IllegalArgumentException: Cannot get property PropertyBool{name=draining, clazz=class java.lang.Boolean, values=[true, false]} as it does not exist in BlockStateContainer{block=minecraft:air, properties=[]}
    at net.minecraft.block.state.BlockStateContainer$StateImplementation.func_177229_b(BlockStateContainer.java:192)
    at org.cyclops.evilcraft.tileentity.TileDarkTank.shouldAutoDrain(TileDarkTank.java:70)
    at org.cyclops.evilcraft.tileentity.TileDarkTank.updateTileEntity(TileDarkTank.java:75)
    at org.cyclops.cyclopscore.tileentity.CyclopsTileEntity.updateTicking(CyclopsTileEntity.java:95)
    at org.cyclops.cyclopscore.tileentity.CyclopsTileEntity.access$000(CyclopsTileEntity.java:38)
    at org.cyclops.cyclopscore.tileentity.CyclopsTileEntity$TickingTileComponent.func_73660_a(CyclopsTileEntity.java:323)
    at org.cyclops.evilcraft.tileentity.TileDarkTank.func_73660_a(TileDarkTank.java:40)
    at net.minecraft.world.World.func_72939_s(World.java:1791)
    at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:619)
    at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:705)
    at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:386)
    at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:609)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:467)
    at java.lang.Thread.run(Thread.java:745)

## A detailed walkthrough of the error, its code path and all known details is as follows:

-- Head --
Stacktrace:
    at net.minecraft.block.state.BlockStateContainer$StateImplementation.func_177229_b(BlockStateContainer.java:192)
    at org.cyclops.evilcraft.tileentity.TileDarkTank.shouldAutoDrain(TileDarkTank.java:70)
    at org.cyclops.evilcraft.tileentity.TileDarkTank.updateTileEntity(TileDarkTank.java:75)
    at org.cyclops.cyclopscore.tileentity.CyclopsTileEntity.updateTicking(CyclopsTileEntity.java:95)
    at org.cyclops.cyclopscore.tileentity.CyclopsTileEntity.access$000(CyclopsTileEntity.java:38)
    at org.cyclops.cyclopscore.tileentity.CyclopsTileEntity$TickingTileComponent.func_73660_a(CyclopsTileEntity.java:323)
    at org.cyclops.evilcraft.tileentity.TileDarkTank.func_73660_a(TileDarkTank.java:40)

-- Block entity being ticked --
Details:
    Name: darkTank // org.cyclops.evilcraft.tileentity.TileDarkTank
    Block type: ID #0 (tile.air // net.minecraft.block.BlockAir)
    Block data value: 0 / 0x0 / 0b0000
    Block location: World: (-505,63,397), Chunk: (at 7,3,13 in -32,24; contains blocks -512,0,384 to -497,255,399), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
    Actual block type: ID #0 (tile.air // net.minecraft.block.BlockAir)
    Actual block data value: 0 / 0x0 / 0b0000
Stacktrace:
    at net.minecraft.world.World.func_72939_s(World.java:1791)
    at net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:619)

-- Affected level --
Details:
    Level name: world
    All players: 2 total; [EntityPlayerMP['R_Thomson'/217, l='world', x=-516.48, y=65.00, z=409.96], EntityPlayerMP['Mr_Altar'/496, l='world', x=-582.30, y=13.00, z=473.69]]
    Chunk stats: ServerChunkCache: 866 Drop: 0
    Level seed: -1704741057
    Level generator: ID 06 - BIOMESOP, ver 0. Features enabled: true
    Level generator options: 
    Level spawn location: World: (-160,64,256), Chunk: (at 0,4,0 in -10,16; contains blocks -160,0,256 to -145,255,271), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
    Level time: 3135298 game time, 3649988 day time
    Level dimension: 0
    Level storage version: 0x04ABD - Anvil
    Level weather: Rain time: 53721 (now: false), thunder time: 39496 (now: false)
    Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
Stacktrace:
    at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:705)
    at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:386)
    at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:609)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:467)
    at java.lang.Thread.run(Thread.java:745)

-- System Details --
Details:
    Minecraft Version: 1.9
    Operating System: Linux (amd64) version 2.6.32-573.7.1.el6.x86_64
    Java Version: 1.8.0_60, Oracle Corporation
    Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
    Memory: 2267317736 bytes (2162 MB) / 2978480128 bytes (2840 MB) up to 2978480128 bytes (2840 MB)
    JVM Flags: 5 total; -Xms3000M -Xmx3000M -XX:+AggressiveOpts -XX:+UseFastAccessorMethods -XX:ParallelGCThreads=4
    IntCache: cache: 0, tcache: 0, allocated: 3, tallocated: 104
    FML: MCP 9.23 Powered by Forge 12.16.1.1889 75 mods loaded, 75 mods active
    States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
    UCHIJAAAA   mcp{9.19} [Minecraft Coder Pack](minecraft.jar) 
    UCHIJAAAA   FML{8.0.99.99} [Forge Mod Loader](forge-1.9-12.16.1.1889-universal.jar) 
    UCHIJAAAA   Forge{12.16.1.1889} [Minecraft Forge](forge-1.9-12.16.1.1889-universal.jar) 
    UCHIJAAAA   Baubles{1.2.0.2} [Baubles](Baubles-1.9-1.2.0.2.jar) 
    UCHIJAAAA   baublesstuff{2.0.7} [Baubles Stuff](baublesstuff-2.0.7.jar) 
    UCHIJAAAA   bdlib{1.11.0.14} [BD Lib](bdlib-1.11.0.14-mc1.9.jar) 
    UCHIJAAAA   BetterAchievements{0.2.5.19} [Better Achievements](BetterAchievements-1.9-0.2.5.19.jar) 
    UCHIJAAAA   BBG{4.0.0} [BetterBedrockGen](BetterBedrockGenerator-1.9.0-3.0.0.jar) 
    UCHIJAAAA   BiomesOPlenty{4.0.1.2006} [Biomes O' Plenty](BiomesOPlenty-1.9-4.0.1.2006-universal.jar) 
    UCHIJAAAA   JEI{3.3.3.198} [Just Enough Items](jei_1.9-3.3.3.198.jar) 
    UCHIJAAAA   BloodMagic{1.9-2.0.0-36} [Blood Magic: Alchemical Wizardry](BloodMagic-1.9-2.0.0-36.jar) 
    UCHIJAAAA   bookshelf{1.2.0.226} [Bookshelf](Bookshelf-1.9-1.2.0.237.jar) 
    UCHIJAAAA   bookshelfapi{3.0.4} [Bookshelf](Bookshelf-API-1.9-3.0.4.jar) 
    UCHIJAAAA   Chameleon{1.9-2.1.3} [Chameleon](Chameleon-1.9-2.1.3.jar) 
    UCHIJAAAA   chancecubes{1.9-2.1.1.98} [Chance Cubes](ChanceCubes-1.9-2.1.1.98.jar) 
    UCHIJAAAA   ChestTransporter{2.3.0} [Chest Transporter](ChestTransporter-2.3.0-MC1.9.jar) 
    UCHIJAAAA   chisel{@VERSION@} [Chisel](Chisel-MC1.9-0.0.4.13.jar) 
    UCHIJAAAA   mcmultipart{1.1.1} [MCMultiPart](MCMultiPart-1.1.1-universal.jar) 
    UCHIJAAAA   chiselsandbits{9.15} [Chisels & Bits](chiselsandbits-9.15.jar) 
    UCHIJAAAA   compactstorage{v2.0.0a} [CompactStorage](compactstorage-2.1-1.9-rc.jar) 
    UCHIJAAAA   cookingforblockheads{2.0.5} [Cooking for Blockheads](CookingForBlockheads-mc1.9-2.0.5.jar) 
    UCHIJAAAA   craftablenametags{1.2} [Craftable Name Tags](craftablenametags-1.2.jar) 
    UCHIJAAAA   craftingtweaks{4.0.10} [Crafting Tweaks](craftingtweaks-mc1.9-4.0.10.jar) 
    UCHIJAAAA   MineTweaker3{3.0.7} [MineTweaker 3](CraftTweaker-Customized-1.9-3.0.7.jar) 
    UCHIJAAAA   cyclopscore{0.5.8} [Cyclops Core](CyclopsCore-1.9-0.5.8.jar) 
    UCHIJAAAA   eplus{4.1.0.86} [Enchanting Plus](EnchantingPlus-1.9-4.1.0.97.jar) 
    UCHIJAAAA   EnderBags{1.0.1} [EnderBags](EnderBags-1.9-1.0.1.jar) 
    UCHIJAAAA   evilcraft{0.9.9} [EvilCraft](EvilCraft-1.9-0.9.9.jar) 
    UCHIJAAAA   excore{1.5.3} [EXCore](Excore-1.5.3-1.9.jar) 
    UCHIJAAAA   extrathings{0.2.0} [Extra Things](extraThings-0.2.0.jar) 
    UCHIJAAAA   ezstorage{1.0} [EZ Storage](ezstorage-1.9-1.2.1.jar) 
    UCHIJAAAA   fastleafdecay{1.6} [Fast Leaf Decay](FastLeafDecay-1.9-1.6.jar) 
    UCHIJAAAA   substratum{1.3.0.0} [Substratum](Substratum-1.9-1.3.0.0.jar) 
    UCHIJAAAA   foundry{2.0.0.0-alpha2} [Foundry](Foundry-1.9-2.0.0.0-alpha2.jar) 
    UCHIJAAAA   furnus{1.74} [Furnus](Furnus-1.9-1.74.jar) 
    UCHIJAAAA   advgenerators{0.9.20.9} [Advanced Generators](generators-0.9.20.9-mc1.9.jar) 
    UCHIJAAAA   graves{1.9-b10-universal} [Graves Mod](graves-1.9-b10-universal.jar) 
    UCHIJAAAA   HardcoreQuesting{5.0.0alpha1} [Hardcore Questing Mode](HQM-New Heights-5.0.0alpha1.jar) 
    UCHIJAAAA   inventorysorter{0.7.14+32} [Simple Inventory sorting](inventorysorter-1.9-0.7.14+32.jar) 
    UCHIJAAAA   IronChest{6.0.133.777} [Iron Chest](ironchest-1.9-6.0.133.777.jar) 
    UCHIJAAAA   journeymap{1.9-5.2.0b1} [JourneyMap](journeymap-1.9-5.2.0b1-unlimited.jar) 
    UCHIJAAAA   LunatriusCore{1.1.2.34} [LunatriusCore](LunatriusCore-1.9-1.1.2.34-universal.jar) 
    UCHIJAAAA   malisiscore{1.9-3.0.6} [MalisisCore](malisiscore-1.9-3.0.6.jar) 
    UCHIJAAAA   malisisblocks{1.9-3.0.1} [Malisis Blocks](malisisblocks-1.9-3.0.1.jar) 
    UCHIJAAAA   malisisdoors{1.9-4.0.1} [MalisisDoors](malisisdoors-1.9-4.0.1.jar) 
    UCHIJAAAA   malisisswitches{1.9-2.0.0} [Malisis Switches](malisisswitches-1.9-2.0.0.jar) 
    UCHIJAAAA   mantle{1.9-0.10.0.jenkins132} [Mantle](Mantle-1.9-0.10.0.jar) 
    UCHIJAAAA   RadixCore{1.9-2.1.0} [RadixCore](RadixCore-1.9-2.1.0-universal.jar) 
    UCHIJAAAA   MCA{1.9-5.1.2} [Minecraft Comes Alive](MCA-1.9-5.1.2-universal.jar) 
    UCHIJAAAA   testdummy{1.10} [MmmMmmMmmMmm](MmmMmmMmmMmm-1.10.jar) 
    UCHIJAAAA   Morpheus{1.9-3.0.2} [Morpheus](Morpheus-1.9-3.0.2.jar) 
    UCHIJAAAA   tconstruct{1.9-2.3.0.jenkins193} [Tinkers' Construct](TConstruct-1.9-2.3.0.jar) 
    UCHIJAAAA   neotech{3.0.6} [NeoTech](NeoTech-1.9-3.0.6.jar) 
    UCHIJAAAA   notenoughwands{1.3.2} [Not Enough Wands](notenoughwands-1.9.0-1.3.2.jar) 
    UCHIJAAAA   progressiveautomation{1.6.42} [Progressive Automation](ProgressiveAutomation-1.9-1.6.42.jar) 
    UCHIJAAAA   Psi{beta-31} [Psi](Psi-beta-31.jar) 
    UCHIJAAAA   quantumflux{2.0.6} [QuantumFlux](quantumflux-1.9-2.0.6.jar) 
    UCHIJAAAA   randomthings{3.7} [Random Things](RandomThings-MC1.9-3.7.jar) 
    UCHIJAAAA   reborncore{2.1.0.53} [RebornCore](RebornCore-1.9-2.1.0.53-universal.jar) 
    UCHIJAAAA   xreliquary{1.9-1.3.1.338} [Reliquary](Reliquary-1.9-1.3.1.338.jar) 
    UCHIJAAAA   rftools{4.46} [RFTools](rftools-1.9.0-4.46.jar) 
    UCHIJAAAA   rftoolsdim{4.23beta37} [RFTools Dimensions](rftoolsdim-1.9.0-4.23beta37.jar) 
    UCHIJAAAA   Roguelike{1.5.0} [Roguelike Dungeons](RoguelikeDungeons-1.9-1.5.0.jar) 
    UCHIJAAAA   shetiphiancore{3.2.1} [ShetiPhian-Core](shetiphiancore-1.9.0-3.2.1.jar) 
    UCHIJAAAA   simpleretrogen{2.2.3+4} [Simple Retrogen](simpleretrogen-1.9-2.2.3+4.jar) 
    UCHIJAAAA   soulshardstow{1.9-2.3.9-25} [Soul Shards - The Old Ways](SoulShards-TOW-1.9-2.3.9-25.jar) 
    UCHIJAAAA   StorageDrawers{1.9-3.1.6} [Storage Drawers](StorageDrawers-1.9-3.1.6.jar) 
    UCHIJAAAA   techreborn{1.0.6.197} [TechReborn](TechReborn-1.9-1.0.6.197-universal.jar) 
    UCHIJAAAA   Torcherino{@VERSION@} [Torcherino](Torcherino-TehNut-1.9-2.2s.jar) 
    UCHIJAAAA   trashslot{3.0.4} [TrashSlot](trashslot-mc1.9-3.0.4.jar) 
    UCHIJAAAA   VeinMiner{0.31.5_build.forgePost18010} [Vein Miner](VeinMiner-1.9_0.31.5.forgePost18010.jar) 
    UCHIJAAAA   VeinMinerModSupport{0.31.5_build.forgePost18010} [Mod Support](VeinMiner-1.9_0.31.5.forgePost18010.jar) 
    UCHIJAAAA   Waila{1.7.0} [Waila](Waila-1.7.0-B2_1.9.jar) 
    UCHIJAAAA   wailafeatures{@MAJOR@.@MINOR@.@REVIS@} [WAILA Features](WAILA-features-1.9-0.3.1.6.jar) 
    UCHIJAAAA   WailaHarvestability{1.1.7} [Waila Harvestability](WailaHarvestability-mc1.9-1.1.7.jar) 
    Loaded coremods (and transformers): 
FldLoadingPlugin (FastLeafDecay-1.9-1.6.jar)
  com.olafski.fastleafdecay.FldTransformer
LoadingPlugin (RandomThings-MC1.9-3.7.jar)
  lumien.randomthings.asm.ClassTransformer
Reborn Core ASM (RebornCore-1.9-2.1.0.53-universal.jar)
  reborncore.asm.RebornClassTransformer
MalisisCorePlugin (malisiscore-1.9-3.0.6.jar)
  net.malisis.core.util.chunkcollision.ChunkCollisionTransformer
  net.malisis.core.util.chunkblock.ChunkBlockTransformer
  net.malisis.core.renderer.transformer.MalisisRendererTransformer
  net.malisis.core.renderer.icon.asm.TextureMapTransformer
MalisisSwitchesPlugin (malisisswitches-1.9-2.0.0.jar)
  net.malisis.switches.asm.MalisisSwitchesTransformer
    Pulsar/tconstruct loaded Pulses: 
        - TinkerCommons (Enabled/Forced)
        - TinkerWorld (Enabled/Not Forced)
        - TinkerTools (Enabled/Not Forced)
        - TinkerSmeltery (Enabled/Not Forced)
        - TinkerGadgets (Enabled/Not Forced)
        - TinkerOredict (Enabled/Forced)
        - TinkerIntegration (Enabled/Forced)
        - TinkerFluids (Enabled/Forced)
        - TinkerMaterials (Enabled/Forced)
        - chiselsandbitsIntegration (Enabled/Not Forced)
        - craftingtweaksIntegration (Enabled/Not Forced)

```
List of loaded APIs: 
    * Baubles|API (1.1.1.0) from Baubles-1.9-1.2.0.2.jar
    * BetterAchievements|API (0.2.5.19) from BetterAchievements-1.9-0.2.5.19.jar
    * BloodMagic|API (@VERSION@) from BloodMagic-1.9-2.0.0-36.jar
    * ChiselAPI (0.0.1) from Chisel-MC1.9-0.0.4.13.jar
    * ChiselAPI|Carving (0.0.1) from Chisel-MC1.9-0.0.4.13.jar
    * ChiselsAndBitsAPI (1.8.1) from chiselsandbits-9.15.jar
    * CoFHAPI (1.8.9R1.2.0B1) from Bookshelf-API-1.9-3.0.4.jar
    * CoFHAPI|block (1.8.9R1.2.0B1) from ProgressiveAutomation-1.9-1.6.42.jar
    * CoFHAPI|energy (1.8.9R1.2.0B1) from Bookshelf-API-1.9-3.0.4.jar
    * CoFHAPI|tileentity (1.8.9R1.2.0B1) from ProgressiveAutomation-1.9-1.6.42.jar
    * compactstorage (v2.0.0a) from compactstorage-2.1-1.9-rc.jar
    * CraftingTweaks|API (4.1) from craftingtweaks-mc1.9-4.0.10.jar
    * foundry.api (2.0.0.0) from Foundry-1.9-2.0.0.0-alpha2.jar
    * IC2API (1.0) from RebornCore-1.9-2.1.0.53-universal.jar
    * jeresources|API (0.4.5.25) from JustEnoughResources-1.9-0.4.5.25.jar
    * journeymap|client-api (1.2-SNAPSHOT) from journeymap-1.9-5.2.0b1-unlimited.jar
    * journeymap|client-api-display (1.2-SNAPSHOT) from journeymap-1.9-5.2.0b1-unlimited.jar
    * journeymap|client-api-event (1.2-SNAPSHOT) from journeymap-1.9-5.2.0b1-unlimited.jar
    * journeymap|client-api-model (1.2-SNAPSHOT) from journeymap-1.9-5.2.0b1-unlimited.jar
    * journeymap|client-api-util (1.2-SNAPSHOT) from journeymap-1.9-5.2.0b1-unlimited.jar
    * JustEnoughItemsAPI (3.1.0) from jei_1.9-3.3.3.198.jar
    * McJtyLib (1.9.0-1.8.3) from mcjtylib-1.9.0-1.8.3.jar
    * PsiAPI (2) from Psi-beta-31.jar
    * reborncoreAPI (2.1.0.53) from RebornCore-1.9-2.1.0.53-universal.jar
    * SoulShardsTOW|API (0.1.5) from SoulShards-TOW-1.9-2.3.9-25.jar
    * StorageDrawersAPI (1.7.10-1.2.0) from StorageDrawers-1.9-3.1.6.jar
    * StorageDrawersAPI|config (1.7.10-1.2.0) from StorageDrawers-1.9-3.1.6.jar
    * StorageDrawersAPI|event (1.7.10-1.2.0) from StorageDrawers-1.9-3.1.6.jar
    * StorageDrawersAPI|inventory (1.7.10-1.2.0) from StorageDrawers-1.9-3.1.6.jar
    * StorageDrawersAPI|pack (1.7.10-1.2.0) from StorageDrawers-1.9-3.1.6.jar
    * StorageDrawersAPI|registry (1.7.10-1.2.0) from StorageDrawers-1.9-3.1.6.jar
    * StorageDrawersAPI|render (1.7.10-1.2.0) from StorageDrawers-1.9-3.1.6.jar
    * StorageDrawersAPI|storage (1.7.10-1.2.0) from StorageDrawers-1.9-3.1.6.jar
    * StorageDrawersAPI|storage-attribute (1.7.10-1.2.0) from StorageDrawers-1.9-3.1.6.jar
    * techrebornAPI (1.0.6.197) from TechReborn-1.9-1.0.6.197-universal.jar
    * VeinMinerApi (0.1) from VeinMiner-1.9_0.31.5.forgePost18010.jar
    * WailaAPI (1.3) from Waila-1.7.0-B2_1.9.jar
Profiler Position: N/A (disabled)
Player Count: 2 / 24; [EntityPlayerMP['R_Thomson'/217, l='world', x=-516.48, y=65.00, z=409.96], EntityPlayerMP['Mr_Altar'/496, l='world', x=-582.30, y=13.00, z=473.69]]
Is Modded: Definitely; Server brand changed to 'fml,forge'
Type: Dedicated Server (map_server.txt)
```

We have done some testting and it seem to be all tanks from evilcraft

**Comment by b0bst3r** [comment]
Any sign of a fix for this please as it crashes server, remove liquid that was in the tank when you place it down and also downgrades the tank back to a 16,000mb tank.

**Comment by rubensworks** [comment]
Fixed in CyclopsMC/CyclopsCore@5a30e2cb88b145c2add8919c3c62236fa18ec95a, release coming soon.

**Comment by kwpugh** [comment]
Awesome!
